test/dist_autograd_test.py

```diff
@@ -3,24 +3,49 @@
 import time
 import unittest
 
+from enum import IntEnum
 import torch
 import torch.distributed.autograd as dist_autograd
 import torch.distributed.rpc as rpc
 from dist_utils import INIT_METHOD_TEMPLATE, dist_init
 
 import threading
 
-prev_rank_rpc_done = False
-prev_rank_context_id = -1
-prev_prev_rank_rpc_done = False
-prev_prev_rank_context_id = -1
+# Right now we test up to 3-layer nested rpc calls.
+# rpc_done[RankDistance.PREV] and ctx_ids[RankDistance.PREV]
+# represent rpc is done in prev rank, and context id sent from prev rank
+# respectively.
+# [RankDistance.PREV_PREV] represents for prev of prev rank.
+# [RankDistance.PREV_PREV_PREV] represents for prev of prev of prev rank.
+class RankDistance(IntEnum):
```

Comment:
Why don't we just use integers for the distance instead of an enum like this? That way we can extend this to multiple nested levels later on easily.

Reply:
I thought enum is more descriptive, but yes, I can change it back to integers for add more levels easily